widget/richtext_objects.go

```diff
@@ -0,0 +1,276 @@
+		TextStyle: fyne.TextStyle{Bold: true},
+	}
+)
+
+// HyperlinkSegment represents a hyperlink within a rich text widget.
+//
+// Since: 2.1
+type HyperlinkSegment struct {
+	Alignment fyne.TextAlign
+	Text      string
+	URL       *url.URL
+}
+
+// Inline returns true as hyperlinks are inside other elements.
+func (h *HyperlinkSegment) Inline() bool {
+	return true
+}
+
+// Textual returns the content of this segment rendered to plain text.
+func (h *HyperlinkSegment) Textual() string {
+	return h.Text
+}
+
+// Visual returns the hyperlink widget required to render this segment.
+func (h *HyperlinkSegment) Visual() fyne.CanvasObject {
+	link := NewHyperlink(h.Text, h.URL)
+	link.Alignment = h.Alignment
+	return &fyne.Container{Layout: &unpadTextWidgetLayout{}, Objects: []fyne.CanvasObject{link}}
+}
+
+// Update applies the current state of this hyperlink segment to an existing visual.
+func (h *HyperlinkSegment) Update(o fyne.CanvasObject) {
+	link := o.(*fyne.Container).Objects[0].(*Hyperlink)
+	link.Text = h.Text
+	link.URL = h.URL
+	link.Alignment = h.Alignment
+	link.Refresh()
+}
+
+// Select tells the segment that the user is selecting the content between the two positions.
+func (h *HyperlinkSegment) Select(begin, end fyne.Position) {
+	// no-op: this will be added when we progress to editor
+}
+
+// SelectedText should return the text representation of any content currently selected through the Select call.
+func (h *HyperlinkSegment) SelectedText() string {
+	// no-op: this will be added when we progress to editor
+	return ""
+}
+
+// Unselect tells the segment that the user is has cancelled the previous selection.
+func (h *HyperlinkSegment) Unselect() {
+	// no-op: this will be added when we progress to editor
+}
+
+// SeparatorSegment includes a horizontal separator in a rich text widget.
+//
+// Since: 2.1
+type SeparatorSegment struct {
+}
+
+// Inline returns false as a separator should be full width.
+func (s *SeparatorSegment) Inline() bool {
+	return false
+}
+
+// Textual returns no content for a separator element.
+func (s *SeparatorSegment) Textual() string {
+	return ""
+}
+
+// Visual returns the separator element for this segment.
+func (s *SeparatorSegment) Visual() fyne.CanvasObject {
+	return NewSeparator()
+}
+
+// Update doesnt need to change a separator visual.
+func (s *SeparatorSegment) Update(fyne.CanvasObject) {
+}
+
+// Select does nothing for a separator.
+func (s *SeparatorSegment) Select(begin, end fyne.Position) {
+}
+
+// SelectedText returns the empty string for this separator.
+func (s *SeparatorSegment) SelectedText() string {
+	return "" // TODO maybe return "---\n"?
+}
+
+// Unselect does nothing for a separator.
+func (s *SeparatorSegment) Unselect() {
+}
+
+// RichTextStyle describes the details of a text object inside a RichText widget.
+//
+// Since: 2.1
+type RichTextStyle struct {
+	Alignment fyne.TextAlign
+	ColorName fyne.ThemeColorName
+	Inline    bool
+	SizeName  fyne.ThemeSizeName
```

Comment:
So if users still want to use a custom size for the text, they need to use `canvas.Text` with no text wrapping and other benefits of the RichText? ☹️

Reply:
> No that is not required, they can add a new size to the theme

😯  didn't know that was possible, how could I add new sizes to the theme?

Reply:
Inside the `Size` method if your theme look for another `ThemeSizeName` that is defined in your app/theme.
If you see it then return your app/theme specific value.

Reply:
Oh you are right. I didn't realize it before. That's really great, thanks :)